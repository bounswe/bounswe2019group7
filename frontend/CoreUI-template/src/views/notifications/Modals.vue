<template>
  <div class="wrapper">
    <div class="animated fadeIn">
      <b-row>
        <b-col cols="12">
          <b-card>
            <div slot="header"><i class='fa fa-align-justify'></i> Bootstrap Modals</div>
            <b-button type="button" variant="secondary" @click="myModal = true" class="mr-1">Launch demo modal</b-button>
            <b-button type="button" variant="secondary" @click="largeModal = true" class="mr-1">Launch large modal</b-button>
            <b-button type="button" variant="secondary" @click="smallModal = true" class="mr-1">Launch small modal</b-button>
            <hr>
            <b-button type="button" variant="primary" @click="primaryModal = true" class="mr-1">Launch primary modal</b-button>
            <b-button type="button" variant="success" @click="successModal = true" class="mr-1">Launch success modal</b-button>
            <b-button type="button" variant="warning" @click="warningModal = true" class="mr-1">Launch warning modal</b-button>
            <b-button type="button" variant="danger" @click="dangerModal = true" class="mr-1">Launch danger modal</b-button>
            <b-button type="button" variant="info" @click="infoModal = true" class="mr-1">Launch info modal</b-button>
          </b-card>
        </b-col><!--/.col-->
      </b-row><!--/.row-->
    </div>

    <!-- Modal Component -->
    <b-modal title="Modal title" v-model="myModal" @ok="myModal = false">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </b-modal>
    <b-modal title="Modal title" size="lg" v-model="largeModal" @ok="largeModal = false">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </b-modal>
    <b-modal title="Modal title" size="sm" v-model="smallModal" @ok="smallModal = false">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </b-modal>



    <b-modal 
      title="Modal title" 
      variant="primary"
      header-bg-variant="primary"
      content-class="border-primary"
      v-model="primaryModal" 
      @ok="primaryModal = false"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </b-modal>
    <b-modal 
      title="Modal title" 
      variant="success"
      header-bg-variant="success"
      content-class="border-success"
      v-model="successModal" 
      @ok="successModal = false" 
      ok-variant="success"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </b-modal>
    <b-modal 
      title="Modal title" 
      variant="warning"
      header-bg-variant="warning"
      content-class="border-warning"
      v-model="warningModal" 
      @ok="warningModal = false" 
      ok-variant="warning"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </b-modal>
    <b-modal 
      title="Modal title" 
      variant="danger"
      header-bg-variant="danger"
      content-class="border-danger" 
      v-model="dangerModal" 
      @ok="dangerModal = false" 
      ok-variant="danger"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </b-modal>
    <b-modal 
      title="Modal title" 
      variant="info"
      header-bg-variant="info"
      content-class="border-info" 
      v-model="infoModal" 
      @ok="infoModal = false" 
      ok-variant="info"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </b-modal>
  </div>

</template>

<script>
export default {
  name: 'modals',
  data () {
    return {
      myModal: false,
      largeModal: false,
      smallModal: false,
      primaryModal: false,
      successModal: false,
      warningModal: false,
      dangerModal: false,
      infoModal: false
    }
  }
}
</script>
