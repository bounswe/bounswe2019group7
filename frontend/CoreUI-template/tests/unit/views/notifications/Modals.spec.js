import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Modals from '@/views/notifications/Modals'

Vue.use(BootstrapVue)

describe('Modals.vue', () => {
  it('has a name', () => {
    expect(Modals.name).toMatch('modals')
  })
  it('has a created hook', () => {
    expect(typeof Modals.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Modals.data).toMatch('function')
    const defaultData = Modals.data()
    expect(defaultData.largeModal).toBe(false)
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Modals)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Modals', () => {
    const wrapper = shallowMount(Modals)
    expect(wrapper.is(Modals)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Modals)
    expect(wrapper.find('div.card-header > div').text()).toMatch('Bootstrap Modals')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Modals)
    expect(wrapper.element).toMatchSnapshot()
  })
})
