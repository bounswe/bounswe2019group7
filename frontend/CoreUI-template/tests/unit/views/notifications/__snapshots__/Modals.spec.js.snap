// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals.vue renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="animated fadeIn"
  >
    <b-row-stub
      tag="div"
    >
      <b-col-stub
        cols="12"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          footertag="div"
          headertag="div"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <div
            slot="header"
          >
            <i
              class="fa fa-align-justify"
            />
             Bootstrap Modals
          </div>
           
          <b-button-stub
            class="mr-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="secondary"
          >
            Launch demo modal
          </b-button-stub>
           
          <b-button-stub
            class="mr-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="secondary"
          >
            Launch large modal
          </b-button-stub>
           
          <b-button-stub
            class="mr-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="secondary"
          >
            Launch small modal
          </b-button-stub>
           
          <hr />
           
          <b-button-stub
            class="mr-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="primary"
          >
            Launch primary modal
          </b-button-stub>
           
          <b-button-stub
            class="mr-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="success"
          >
            Launch success modal
          </b-button-stub>
           
          <b-button-stub
            class="mr-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="warning"
          >
            Launch warning modal
          </b-button-stub>
           
          <b-button-stub
            class="mr-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="danger"
          >
            Launch danger modal
          </b-button-stub>
           
          <b-button-stub
            class="mr-1"
            event="click"
            routertag="a"
            tag="button"
            target="_self"
            type="button"
            variant="info"
          >
            Launch info modal
          </b-button-stub>
        </b-card-stub>
      </b-col-stub>
    </b-row-stub>
  </div>
   
  <b-modal-stub
    buttonsize=""
    canceltitle="Cancel"
    cancelvariant="secondary"
    headercloselabel="Close"
    oktitle="OK"
    okvariant="primary"
    size="md"
    title="Modal title"
    titletag="h5"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </b-modal-stub>
   
  <b-modal-stub
    buttonsize=""
    canceltitle="Cancel"
    cancelvariant="secondary"
    headercloselabel="Close"
    oktitle="OK"
    okvariant="primary"
    size="lg"
    title="Modal title"
    titletag="h5"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </b-modal-stub>
   
  <b-modal-stub
    buttonsize=""
    canceltitle="Cancel"
    cancelvariant="secondary"
    headercloselabel="Close"
    oktitle="OK"
    okvariant="primary"
    size="sm"
    title="Modal title"
    titletag="h5"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </b-modal-stub>
   
  <b-modal-stub
    buttonsize=""
    canceltitle="Cancel"
    cancelvariant="secondary"
    class="modal-primary"
    headercloselabel="Close"
    oktitle="OK"
    okvariant="primary"
    size="md"
    title="Modal title"
    titletag="h5"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </b-modal-stub>
   
  <b-modal-stub
    buttonsize=""
    canceltitle="Cancel"
    cancelvariant="secondary"
    class="modal-success"
    headercloselabel="Close"
    oktitle="OK"
    okvariant="success"
    size="md"
    title="Modal title"
    titletag="h5"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </b-modal-stub>
   
  <b-modal-stub
    buttonsize=""
    canceltitle="Cancel"
    cancelvariant="secondary"
    class="modal-warning"
    headercloselabel="Close"
    oktitle="OK"
    okvariant="warning"
    size="md"
    title="Modal title"
    titletag="h5"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </b-modal-stub>
   
  <b-modal-stub
    buttonsize=""
    canceltitle="Cancel"
    cancelvariant="secondary"
    class="modal-danger"
    headercloselabel="Close"
    oktitle="OK"
    okvariant="danger"
    size="md"
    title="Modal title"
    titletag="h5"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </b-modal-stub>
   
  <b-modal-stub
    buttonsize=""
    canceltitle="Cancel"
    cancelvariant="secondary"
    class="modal-info"
    headercloselabel="Close"
    oktitle="OK"
    okvariant="info"
    size="md"
    title="Modal title"
    titletag="h5"
  >
    
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
    quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
    consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
    cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  
  </b-modal-stub>
</div>
`;
